### PR TITLE
opencollada: include pcre.h (KhronosGroup/OpenCOLLADA#570)

### PR DESCRIPTION
**Pull request by eadwu**
###### Motivation for this change
Fixes build (or at least on nixOS)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bendlas** [comment]
attached patch fails against latest master

```
building '/nix/store/8fpiy3j76r10zs33r7ixp87s3n6d03xa-opencollada-1.6.63.drv'...
unpacking sources
unpacking source archive /nix/store/r1hcnhfaa8yjpxk6np664qypsh1c2aa5-source
source root is source
patching sources
applying patch /nix/store/baqax7h47drcc4iy2dn9kgyklad5pxn1-pcre.patch
patching file COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h
Hunk #1 FAILED at 3 (different line endings).
Hunk #2 FAILED at 13 (different line endings).
Hunk #3 FAILED at 25 (different line endings).
Hunk #4 FAILED at 36 (different line endings).
4 out of 4 hunks FAILED -- saving rejects to file COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h.rej
builder for '/nix/store/8fpiy3j76r10zs33r7ixp87s3n6d03xa-opencollada-1.6.63.drv' failed with exit code 1
cannot build derivation '/nix/store/l82m8j90a0wd78z9y0hs6g6sxdgn34rj-blender-2.79b.drv': 1 dependencies couldn't be built
error: build of '/nix/store/l82m8j90a0wd78z9y0hs6g6sxdgn34rj-blender-2.79b.drv' failed
```

Looking at it, there seem to be some superflous whitespace changes in the attached patch. Maybe the merge conflict can be resolved, by just adding the PCRE include?

Could you link the original ticket(s), for reference?

**Comment by matthewbauer** [comment]
@grahacofborg build opencollada

**Comment by eadwu** [comment]
Does the build on darwin build without this PR? Don't have a machine with macOS currently installed so can't really test darwin builds.

**Comment by matthewbauer** [comment]
Yeah macOS is broken too so it won't be a blocker.

File: pkgs/development/libraries/opencollada/default.nix [hunk 1]
```diff
@@ -21,6 +21,8 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  patches = [ ./pcre.patch ];
+
   patchPhase = lib.optionalString stdenv.isDarwin ''
```
Comment (matthewbauer) [inline pkgs/development/libraries/opencollada/default.nix]: @eadwu Change this to postPatch and it should fix macOS building.

Comment (eadwu) [reply pkgs/development/libraries/opencollada/default.nix]: Oh didn't see patchPhase, just added it to patchPhase

**Comment by matthewbauer** [comment]
Thanks!

**Pull request merged**